File: leetcode/golang/876.链表的中间节点.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
 func middleNode(head *ListNode) *ListNode {
	n := 0;
	tmp := head;
	for tmp != nil {
		n += 1;
		tmp = tmp.Next;
	}
	k := n / 2;
	ret := head;
	for k > 0 {
		ret = ret.Next;
		k -= 1;
	}
	return ret;
}

File: leetcode/python/876.链表的中间节点.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: ListNode) -> ListNode:
        tmp = head
        n = 0
        while tmp :
            n += 1
            tmp = tmp.next
        k = n // 2

        ret = head
        while k != 0:
            ret = ret.next
            k -= 1
        return ret